code/components/citizen-resources-core/src/StateBagComponent.cpp

```diff
@@ -619,10 +620,36 @@ void StateBagComponentImpl::HandlePacket(int source, std::string_view dataRaw)
 
 	if (!bag)
 	{
+#ifdef IS_FXSERVER
+		if (strncmp(bagName.data(), "entity:", 7) == 0)
+		{
+			try
+			{
+				int entityId = std::stoi(bagName.data() + 7);
+
+				uint32_t owner = m_gameInterface->GetEntityOwner(entityId);
+				if (owner == uint32_t(-1))
+				{
+					return entityId;
```

Comment:
If this case represents 'deferred creation', it might make more sense instead of a return value to have another (optional) callout in the game interface to 'store for later', or similarly, store it _oneself_ in case such a bag gets created.

Reply:
Agreed, also on the above comments about this approach.

On the latter: storing it *oneself* would also mean that systems using this system *must* call its clean-up function to get rid of outdated updates, I normally don't do this by design/principle (make it hard to use it incorrectly). Unless, of course, we turn the State Bag System agnostic of all the other systems or if we can tell the State Bag system it can store packets for later. Your thoughts?

Reply:
The queue being _internal_ seems like a valid idea, too, so any consumers wouldn't have to deal with added complexity other than being able to approve a pending change for later.